### DISCUSSION TITLE
Modifying the BOM for a Specific Build

### DISCUSSION THREAD

**Discussion by palagap** [inventree/InvenTree#3480] [Q&A] [answered]
Hi there,

I’m trying to figure out a way to modify a completed assembly to add new parts to it (e.g. a retrofit). I see that this as happening in a couple of ways:

1. Being able to directly modify a build’s assembly by adding new parts (I don’t think this is possible as the assembly is locked once the build is completed). The whole assembly would have to be changed, which might cause issues with other builds that use the same assembly but don’t need the retrofit. 

2. Creating a completely new assembly with the new part(s) and then transferring the stock from the old assembly to the new one. I’m not sure if this is possible, though.

Here is an example of what I’m trying to accomplish, using the InvenTree demo:

<img width="1695" alt="Screen Shot 2022-08-05 at 12 09 55 PM" src="https://user-images.githubusercontent.com/40808444/183162048-2e56219f-ac68-4d5d-8b3c-d9cbfef5573a.png">

First, I make an example build - in this case a Blue Chair.

<img width="1682" alt="Screen Shot 2022-08-05 at 12 13 32 PM" src="https://user-images.githubusercontent.com/40808444/183162240-1dd0de0b-fba9-4e00-a529-3abdcfd1340b.png">

Then, I can allocate parts to it, based on the BOM.

<img width="1250" alt="Screen Shot 2022-08-05 at 12 17 40 PM" src="https://user-images.githubusercontent.com/40808444/183162549-3d6f63b4-b38e-4532-a3dd-3eac989c477a.png">

However, I can’t add a new part (e.g. a cushion) to the build, because the stock item isn’t already part of the BOM the assembly is based on.

On the other hand, I could duplicate the assembly and add a new part (thereby creating a new BOM); however, I don’t know of a way to “migrate” the parts from the existing build to the new assembly. In other words, I’d like to keep the parts already allocated to BO0019 (Test Blue Chair), while just needing to allocate the new part (e.g. a cushion). Is there a way to accomplish this?

Thanks.

**Comment by matmair** [comment, marked as answer]
Short answer: No, there is not.

You can install extra stock into a certain part but not in the way you described. The whole point of a BOM is to define the parts and then build the assembly so it is the same every time. Retrofitting (ie changing the parts-list) is not in the design requirements rn.

What you could do is to create a variant that has the new BOM and then change the stock item to that variant.

**Reply by palagap** [reply]
I see, thanks for the guidance - I'll try the variant approach and see if that works.

**Reply by matmair** [reply]
Please mark the answer once you had a change to try it out.
